nx/lib/nx.ex

```diff
@@ -16875,6 +16875,65 @@ defmodule Nx do
     divide(log(tensor), log(base))
   end
 
+  @doc """
+  Replaces every value in `tensor` with `value`.
+
+  The returned tensor has the same shape, names and vectorized axes
+  as the given one. The type defaults to the type of `tensor`, however
+  type promotions apply.
+
+  ## Examples
+
+      iex> tensor = Nx.iota({2, 2})
+      iex> Nx.fill(tensor, 5)
+      #Nx.Tensor<
+        s64[2][2]
+        [
+          [5, 5],
+          [5, 5]
+        ]
+      >
+
+      iex> tensor = Nx.iota({2, 2}) |> Nx.vectorize(:x)
+      iex> Nx.fill(tensor, 5)
+      #Nx.Tensor<
+        s64[x: 2][2]
+        [
+          [5, 5],
+          [5, 5]
+        ]
+      >
+
+  ### Type promotions
```

Comment:
This matches the behaviour of `indexed_put` and `put_slice`, which I believe makes sense. Semantically it's the same as putting the value at every index.